limiter/limiter.go

```diff
@@ -0,0 +1,54 @@
+package limiter
+
+import "time"
+
+type LeakyBucket struct {
+	bucket chan struct{}
+	quit   chan struct{}
+}
+
+func (rl *LeakyBucket) fill(burst int) {
+	for i := 0; i < burst; i++ {
+		rl.bucket <- struct{}{}
+	}
+}
+
+func (rl *LeakyBucket) drip(drip float64) {
+	driptime := float64(1000000000) / drip
```

Comment:
I'm not quite sure I follow this math; should it be using `time.Second`?

It might make it clearer to convert to a `time.Duration` here instead of in the `time.Sleep` call?